### PR TITLE
Add ir.imansalmani.IPlan

### PR DESCRIPTION
**Pull request by iman-salmani**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/iman-salmani/iplan**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: ir.imansalmani.IPlan.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          "tag": "1.0.0",
+          "commit": "36e41b5dd3c10e76b0b332ceaef07e29481e50a6",
+          "url": "https://github.com/iman-salmani/iplan.git"
+        }
+      ]
+    }
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin",
+    "build-args": ["--share=network"],
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:45]: the build need to be offline. Either you use vendored crate, or you use the https://github.com/flatpak/flatpak-builder-tools

Also `build-options` is usually above the modules in the manifest.

File: ir.imansalmani.IPlan.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "modules": [
+    {
+      "name": "iplan",
+      "config-opts": ["-Dprofile=default"],
+      "builddir": true,
+      "buildsystem": "meson",
+      "sources": [
+        {
+          "type": "git",
+          "branch": "main",
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:35]: NEVER set a branch.

File: ir.imansalmani.IPlan.json (hunk 1 above)
Comment (iman-salmani) [reply ir.imansalmani.IPlan.json]: i added dependencies by flatpak-cargo-generator but how its works when cargo runs with --offline option in meson?

File: ir.imansalmani.IPlan.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "/share/doc",
+    "/share/gtk-doc",
+    "/share/man",
+    "/share/pkgconfig",
+    "*.la",
+    "*.a"
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin",
+    "build-args": ["--share=network"],
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:28]: this option is not allowed and WILL fail the build.

File: ir.imansalmani.IPlan.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "env": {
+      "RUST_BACKTRACE": "1",
+      "RUST_LOG": "iplan=debug"
+    }
+  },
+  "modules": [
+    {
+      "name": "iplan",
+      "config-opts": ["-Dprofile=default"],
+      "builddir": true,
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:38]:
```suggestion
```

this is default on meson

File: ir.imansalmani.IPlan.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
+{
+  "app-id": "ir.imansalmani.IPlan",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "43",
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:4]:
```suggestion
  "runtime-version": "44",
```

44 is out now.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
changes required to build.

**Comment by iman-salmani** [comment]
hi i changed meson to use archive.

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

File: ir.imansalmani.IPlan.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  ],
+  "modules": [
+    {
+      "name": "iplan",
+      "config-opts": ["-Dprofile=default"],
+      "buildsystem": "meson",
+      "sources": [
+        {
+            "type" : "archive",
+            "url" : "https://github.com/iman-salmani/iplan/releases/download/1.0.0/iplan-1.0.tar.gz"
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:34]: this is missing a checksum `sha256`.

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

**Comment by hfiguiere** [comment]
you removed the build-option that set the path to the rust compiler.

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

File: ir.imansalmani.IPlan.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    "/man",
+    "/share/doc",
+    "/share/gtk-doc",
+    "/share/man",
+    "/share/pkgconfig",
+    "*.la",
+    "*.a"
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin",
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:27]:
```suggestion
    "append-path": "/usr/lib/sdk/rust-stable/bin"
```

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

**Comment by iman-salmani** [comment]
metainfo updated.
i used git repo as a source module in archive like ambrol project #2997.
so should i also update the archive for updates or every time main branch in git repo get updates flathub will notice and reviews the changes?

**Comment by barthalion** [comment]
The appdata file needs some changes:

```
builddir/files/share/appdata/ir.imansalmani.IPlan.appdata.xml: FAILED:
• file-invalid          : <screenshot> failed to load [https://github.com/iman-salmani/iplan/blob/ee9ab7a6adc2afcb6d2440d2af62d580b097427f/data/screenshots/window.png]
• style-invalid         : <caption> cannot end in '.' [The IPlan main window.]
• tag-missing           : <release> required
• style-invalid         : Content before <ul> is too short [8], at least 20 characters required
```

**Comment by barthalion** [comment]
> so should i also update the archive for updates or every time main branch in git repo get updates flathub will notice and reviews the changes?

Once all issues are resolved and your app accepted, you will be responsible for keeping the manifest up to date. There are tools like https://github.com/flathub/flatpak-external-data-checker/ that automate some of this.

File: ir.imansalmani.IPlan.json [hunk 8]
```diff
@@ -35,7 +35,7 @@
         {
             "type" : "archive",
             "url" : "https://github.com/iman-salmani/iplan/releases/download/1.0.0/iplan-1.0.tar.gz",
-            "sha256" : "5165cb6a8b01a21613373e5e73b2ce39fac86348e8cbff50844082c613221301"
+            "sha256" : "5f312c2edfc2033c324db149814cd2ae286c4ab12a466c913ee29fa571cfc9e8"
```
Comment (hfiguiere) [inline ir.imansalmani.IPlan.json:38]: and as a side note, changing the archive without changing name is considered bad practice.

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

**Comment by hfiguiere** [comment]
bot, build ir.imansalmani.IPlan

**Comment by iman-salmani** [comment]
thanks for your patience.

**Pull request closed without merging**